u8timeseries/timeseries.py

```diff
@@ -804,6 +810,17 @@ def _combine_from_pd_ops(self, other: 'TimeSeries',
         conf_hi = self._combine_or_none(self._confidence_hi, other.conf_hi_pd_series(), combine_fn)
         return TimeSeries(series, conf_lo, conf_hi)
 
+    @staticmethod
+    def _fill_missing_dates(series: pd.Series) -> pd.Series:
+        date_axis = series.index
+        size = 3
+        available_frequencies = [date_axis[x:x + size].inferred_freq for x in range(len(date_axis) - size + 1)]
+        available_frequencies = set(filter(None.__ne__, available_frequencies))
+
+        raise_if_not(len(available_frequencies) == 1, 'Could not infer explicit frequency.', logger)
+        suggested_frequency = available_frequencies.pop()
+        return series.resample(suggested_frequency).sum()
```

Comment:
I played a bit with the current version and it seems that the interpolation is not really what one would expect (with some 0 values inserted). Perhaps here you could follow a 2 steps approach. 1st: fill in the missing dates with `np.nan`, and 2nd: interpolate the missing values, using for instance https://github.com/unit8co/u8timeseries/blob/2139a0f40d032b936c76158b01f6f48656567a0c/u8timeseries/utils/missing_values.py#L41 or `pd.Series.interpolate()` directly.
![image](https://user-images.githubusercontent.com/4806678/81603571-d982de80-93ce-11ea-8595-9274ccb8e085.png)

Reply:
I was thinking that we need to fill blanks and lib will automatically interpolate it or leave it to user

Reply:
I think that's a good idea, maybe leave it to the user if an additional flag is passed? But I believe it would still be good to fill the missing timestamps with NaNs, since otherwise the TimeSeries instance cannot be created in the first place, and the user cannot call the missing_value functions either.

Reply:
Whichever way, would be best to avoid default/automatic behavior that the user is probably unaware of (despite it possibly being written in the docs); if allowing it, log it.